crates/oxc_regular_expression/src/options.rs

```diff
@@ -1,26 +1,34 @@
+use crate::ast::RegularExpressionFlags;
+
 #[derive(Clone, Copy, Debug, Default)]
 pub struct ParserOptions {
     /// Used to adjust Span positions to fit the global source code.
     pub span_offset: u32,
-    /// Unicode mode(`u` or `v` flag) enabled or not.
-    pub unicode_mode: bool,
-    /// Extended Unicode mode(`v` flag) enabled or not.
-    pub unicode_sets_mode: bool,
 }
 
 impl ParserOptions {
     #[must_use]
-    pub fn with_span_offset(self, span_offset: u32) -> ParserOptions {
-        ParserOptions { span_offset, ..self }
+    pub fn with_span_offset(self, span_offset: u32) -> Self {
+        Self { span_offset }
     }
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct PatternParserOptions {
+    /// Used to adjust Span positions to fit the global source code.
+    pub span_offset: u32,
+    /// Regular expression flags
+    pub flags: RegularExpressionFlags,
+}
 
+impl PatternParserOptions {
     #[must_use]
-    pub fn with_unicode_mode(self) -> ParserOptions {
-        ParserOptions { unicode_mode: true, ..self }
+    pub fn with_span_offset(self, span_offset: u32) -> Self {
+        Self { span_offset, ..self }
     }
 
     #[must_use]
-    pub fn with_unicode_sets_mode(self) -> ParserOptions {
-        ParserOptions { unicode_mode: true, unicode_sets_mode: true, ..self }
+    pub fn with_flags(self, flags: RegularExpressionFlags) -> Self {
+        Self { flags, ..self }
```

Comment:
I prefer this in order to split the domain more explicit. 👀 

```suggestion
        Self {
            unicode_mode: flags.intersects(RegularExpressionFlags::U | RegularExpressionFlags::V),
            unicode_sets_mode: flags.contains(RegularExpressionFlags::V),
            ..self
        }
```